src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy

```diff
@@ -26,35 +33,42 @@ import static org.openkilda.testing.Constants.NON_EXISTENT_FLOW_ID
 
 class FlowValidationNegativeSpec extends HealthCheckSpecification {
 
+    @Autowired
+    @Shared
+    FlowFactory flowFactory
+    @Autowired
+    @Shared
+    SwitchRulesFactory switchRulesFactory
+
+
     @IterationTag(tags = [SMOKE], iterationNameRegex = /reverse/)
     def "Flow and switch validation should fail in case of missing rules with #flowConfig configuration [#flowType]"() {
         given: "Two flows with #flowConfig configuration"
-        def flowToBreak = flowHelperV2.randomFlow(switchPair, false)
-        def intactFlow = flowHelperV2.randomFlow(switchPair, false, [flowToBreak])
-
-        flowHelperV2.addFlow(flowToBreak)
-        flowHelperV2.addFlow(intactFlow)
+        def flowToBreak = flowFactory.getRandom(switchPair, false)
+        def intactFlow = flowFactory.getRandom(
+                switchPair, false, UP, flowToBreak.occupiedEndpoints())
 
         and: "Both flows have the same switches in path"
-        def damagedFlowSwitches = pathHelper.getInvolvedSwitches(flowToBreak.flowId)*.dpId
-        def intactFlowSwitches = pathHelper.getInvolvedSwitches(intactFlow.flowId)*.dpId
-        assert damagedFlowSwitches.equals(intactFlowSwitches)
+        def damagedFlowSwitches = flowToBreak.retrieveAllEntityPaths().getInvolvedSwitches()
+        def intactFlowSwitches = intactFlow.retrieveAllEntityPaths().getInvolvedSwitches()
+        assert damagedFlowSwitches == intactFlowSwitches
 
         when: "#flowType flow rule from first flow on #switchNo switch gets deleted"
         def cookieToDelete = flowType == "forward" ? database.getFlow(flowToBreak.flowId).forwardPath.cookie.value :
                 database.getFlow(flowToBreak.flowId).reversePath.cookie.value
         SwitchId damagedSwitch = damagedFlowSwitches[item]
-        switchHelper.deleteSwitchRules(damagedSwitch, cookieToDelete)
+        def swRules = switchRulesFactory.get(damagedSwitch)
+        swRules.delete(cookieToDelete)
 
         then: "Intact flow should be validated successfully"
-        def intactFlowValidation = northbound.validateFlow(intactFlow.flowId)
+        def intactFlowValidation = intactFlow.validate()
         intactFlowValidation.each { direction ->
             assert direction.discrepancies.empty
             assert direction.asExpected
         }
```

Comment:
Could you please replace validation response verification by 
`validateAndCollectDiscrepancies()` where is applicable.

Reply:
Done